### ISSUE TITLE
ESM import doesn't work for @casl/mongoose

### ISSUE THREAD

**Issue by Mihailoff** [stalniy/casl#487]
**Describe the bug**
ESM import doesn't work. Most likely because of the unmet peer dependency.

```
file:///usr/src/app/node_modules/@casl/mongoose/dist/es6m/index.mjs:1
app_1        | import{ForbiddenError as t,getDefaultErrorMessage as n,wrapArray as o}from"@casl/ability";import{Query as e}from"mongoose";import{rulesToQuery as r,permittedFieldsOf as i}from"@casl/ability/extra";function c(t){const n=t.conditions;return t.inverted?{$nor:[n]}:n}function s(t,n,o){return r(t,o||"read",n,c)}function f(o,e,r,i){const c=t.from(o);c.action=e;c.subjectType=r;c.setMessage(n(c));i.where({__forbiddenByCasl__:1});i.exec=function t(...n){const o="function"===typeof n[0]?n[0]:n[1];if("function"===typeof o){process.nextTick((()=>o(c)));return}return Promise.reject(c)};return i}function u(t,n){let o=this.modelName;if(!o)o="model"in this?this.model.modelName:null;if(!o)throw new TypeError("Cannot detect model name to return accessible records");const r=s(t,o,n);if(null===r)return f(t,n||"read",o,this.where());return this instanceof e?this.and([r]):this.where({$and:[r]})}function l(t){t.query.accessibleBy=u;t.statics.accessibleBy=u}const a=t=>Object.keys(t.paths);function d(t,n){const e=n.getFields(t);if(!n||!("except"in n))return e;const r=o(n.except);return e.filter((t=>-1===r.indexOf(t)))}function m(){let t;return(n,e)=>{if(!t){const r=e&&"only"in e?o(e.only):d(n,e);t=t=>t.fields||r}return t}}function h(t,n){const o=Object.assign({getFields:a},n);const e=m();function r(n,r){const c="function"===typeof this?this.modelName:this;return i(n,r||"read",c,{fieldsFrom:e(t,o)})}t.statics.accessibleFieldsBy=r;t.method("accessibleFieldsBy",r)}export{h as accessibleFieldsPlugin,l as accessibleRecordsPlugin,a as getSchemaPaths,s as toMongoQuery};
app_1        |                                                                                                  ^^^^^
app_1        | SyntaxError: Named export 'Query' not found. The requested module 'mongoose' is a CommonJS module, which may not support all module.exports as named exports.
app_1        | CommonJS modules can always be imported via the default export, for example using:
app_1        | 
app_1        | import pkg from 'mongoose';
app_1        | const {ForbiddenError: t,getDefaultErrorMessage: n,wrapArray: o}from"@casl/ability";import{Query: e}from"mongoose";import{rulesToQuery: r,permittedFieldsOf: i}from"@casl/ability/extra";function c(t){const n=t.conditions;return t.inverted?{$nor:[n]}:n}function s(t,n,o){return r(t,o||"read",n,c)}function f(o,e,r,i){const c=t.from(o);c.action=e;c.subjectType=r;c.setMessage(n(c));i.where({__forbiddenByCasl__:1});i.exec=function t(...n){const o="function"===typeof n[0]?n[0]:n[1];if("function"===typeof o){process.nextTick((()=>o(c)));return}return Promise.reject(c)};return i}function u(t,n){let o=this.modelName;if(!o)o="model"in this?this.model.modelName:null;if(!o)throw new TypeError("Cannot detect model name to return accessible records");const r=s(t,o,n);if(null===r)return f(t,n||"read",o,this.where());return this instanceof e?this.and([r]):this.where({$and:[r]})}function l(t){t.query.accessibleBy=u;t.statics.accessibleBy=u}const a=t=>Object.keys(t.paths);function d(t,n){const e=n.getFields(t);if(!n||!("except"in n))return e;const r=o(n.except);return e.filter((t=>-1===r.indexOf(t)))}function m(){let t;return(n,e)=>{if(!t){const r=e&&"only"in e?o(e.only):d(n,e);t=t=>t.fields||r}return t}}function h(t,n){const o=Object.assign({getFields:a},n);const e=m();function r(n,r){const c="function"===typeof this?this.modelName:this;return i(n,r||"read",c,{fieldsFrom:e(t,o)})}t.statics.accessibleFieldsBy=r;t.method("accessibleFieldsBy",r)}export{h: accessibleFieldsPlugin,l: accessibleRecordsPlugin,a: getSchemaPaths,s: toMongoQuery} = pkg;
app_1        | 
app_1        |     at ModuleJob._instantiate (internal/modules/esm/module_job.js:104:21)
app_1        |     at async ModuleJob.run (internal/modules/esm/module_job.js:149:5)
app_1        |     at async Loader.import (internal/modules/esm/loader.js:166:24)
app_1        |     at async Object.loadESM (internal/process/esm_loader.js:68:5)
```

**To Reproduce**
1. `npm install --save @casl/mongoose @casl/ability`
2. `import { toMongoQuery } from '@casl/mongoose'`
3. `npm list`

`npm ERR! peer dep missing: mongoose@^4.0.0 || <= 5.10.0, required by @casl/mongoose@4.0.1`

**Expected behaviour**
`toMongoQuery` to load

**CASL Version**

`@casl/ability` - v5.2.2
`@casl/mongoose` - v4.0.1
`mongoose` - v5.12.2 <-------

**Environment**:
Node v14.15.3 (Docker image node:14-stretch-slim)
 
package.json
```JSON
{ "type": "module" }
```

**Comment by Mihailoff** [comment]
`npm install --save mongoose@5.10.0` does not help

**Comment by stalniy** [comment]
Thanks for the issue. I will double check the rollup config to fix it

**Comment by stalniy** [comment]
fixed in `@casl/mongose@4.0.2`. Thanks for the report!
